tools/config/__init__.py

```diff
@@ -710,8 +710,11 @@ def _generate_bootloader_build(self, rom_start, rom_size):
             if part.minaddr() != rom_start:
                 raise ConfigException("bootloader executable does not "
                                       "start at 0x%x" % rom_start)
-            part_size = (part.maxaddr() - part.minaddr()) + 1
+
+            # segments returns start address and 'next after end' address
+            part_size = part.segments()[0][1] - part.segments()[0][0]
```

Comment:
The logic needs to be extended here. s132_nrf52_5.0.0_softdevice.hex contains multiple segments that write into flash address space. It is not enough to do it like this, as the next segment from softdevice hex will collide with the start address we are extracting here.

Reply:
Thanks @juhoeskeli for the review, @JammuKekkonen Please review